src/backend/input.rs

```diff
@@ -0,0 +1,31 @@
+use std::hash::Hash;
+
+#[derive(Hash, PartialEq, Eq, Clone, Copy)]
+pub enum InputDevice<T: PartialEq + Eq + Hash + Clone + Copy>
```

Comment:
The wayland protocol explicitly support multiple WlSeat, and my feeling is that libinput and the wayland protocol have been specifically developped together such that 1 libinput seat = 1 wl_seat.

Seats can then be created and destroyed as devices are plugged/unplugged, according to the udev configuration of the user. I believe the default one is probably "all is merged into a single seat", so from smithay's point of view, plugging a 2nd keyboard would be invisible (as it would be merged with the already existing one).

However, with some specific udev config, plugging a keyboard could trigger the creation of a 2nd seat, yeah.